pkg/security/podsecuritypolicy/provider.go

```diff
@@ -259,7 +259,7 @@ func (s *simpleProvider) validatePodVolumes(pod *api.Pod) field.ErrorList {
 				continue
 			}
 
-			if !allowsAllVolumeTypes && !allowedVolumes.Has(string(fsType)) {
+			if !allowsAllVolumeTypes && !allowedVolumes.Has(string(fsType)) && !(allowedVolumes.Has(string(policy.Secret)) && fsType == policy.Projected && utilfeature.DefaultFeatureGate.Enabled(features.BoundServiceAccountTokenVolume)) {
```

Comment:
I'd recommend splitting this out into a function and not checking the feature gate (if BoundServiceAccountTokenVolume is not enabled, they won't be able to create a pod that uses a BoundServiceAccountTokenVolume):


```go
if !allowsAllVolumeTypes && !allowsVolumeType(allowedVolumes, string(fsType)) {
  ...
```

```go
func allowsVolumeType(allowedVolumes sets.String, fsType string, volume api.Volume) bool {
  if allowedVolumes.Has(fsType) { 
    return true 
  }
  if allowedVolumes.Has(policy.Secret) && fsType == policy.Projected { 
    return true 
  }
  return false
}
```

Reply:
projected volumes include other volume sources like configMap, secret and downwardAPI. if  `BoundServiceAccountTokenVolume` is not enabled, they might want to disallow projected volumes and there should be no special casing.